app/src/main/java/com/ataulm/whatsnext/WhatsNextService.java

```diff
@@ -79,6 +87,50 @@ public Film apply(ApiFilm apiFilm, ApiFilmRelationship apiFilmRelationship) {
                         );
                     }
                 }
-        ).toObservable();
+        )
+                .toObservable()
+                .compose(this.<Film>refreshAccessTokenIfNecessary());
```

Comment:
This needs to go on requests which use the access token. At the moment, those requests aren't well identified -> the interceptor which adds the access token determines for itself whether it's necessary.

I think ideally, there'd be an annotation on the endpoint declaration, and then the interceptors would use the presence of this annotation (`@RequiresAccessToken` for example) to:

- recognise that this is a request that needs an access token instead of just a signature
- intercept missing/invalid access token requests, fetch another one (where possible), and retry the original request

this class doesn't need to worry about it at all.

Reply:
https://github.com/ataulm/whats-next/issues/13